neuralpredictors/layers/readouts/multi_readout.py

```diff
@@ -1,107 +1,147 @@
+        base_readout=None,
+        mean_activity_dict=None,
+        clone_readout=False,
+        gamma_readout=1.0,
+        **kwargs
+    ):
+
+        # The `base_readout` can be overridden only if the static property `_base_readout` is not set
         if self._base_readout is None:
-            raise ValueError("Attribute _base_readout must be set")
+            self._base_readout = base_readout
 
+        if self._base_readout is None:
+            raise ValueError("Attribute _base_readout must be set")
         super().__init__()
 
-        self.in_shape = in_shape
-        self.neurons = OrderedDict([(k, loader.dataset.n_neurons) for k, loader in loaders.items()])
-        if "positive" in kwargs:
-            self._positive = kwargs["positive"]
+        for i, data_key in enumerate(n_neurons_dict):
+            first_data_key = data_key if i == 0 else first_data_key
+            mean_activity = mean_activity_dict[data_key] if mean_activity_dict is not None else None
 
-        self.gamma_readout = gamma_readout  # regularisation strength
+            readout_kwargs = self.prepare_readout_kwargs(i, data_key, first_data_key, **kwargs)
 
-        for i, (k, n_neurons) in enumerate(self.neurons.items()):
             if i == 0 or clone_readout is False:
                 self.add_module(
-                    k,
-                    self._base_readout(in_shape=in_shape, outdims=n_neurons, **kwargs),
+                    data_key,
+                    self._base_readout(
+                        in_shape=in_shape_dict[data_key],
+                        outdims=n_neurons_dict[data_key],
+                        mean_activity=mean_activity,
+                        reg_weight=gamma_readout,
+                        **readout_kwargs
+                    ),
                 )
-                original_readout = k
+                original_readout = data_key
             elif i > 0 and clone_readout is True:
-                self.add_module(k, ClonedReadout(self[original_readout], **kwargs))
-
-    def initialize(self, mean_activity_dict):
-        for k, mu in mean_activity_dict.items():
-            self[k].initialize()
-            if hasattr(self[k], "bias"):
-                self[k].bias.data = mu.squeeze() - 1
-
-    def regularizer(self, readout_key):
-        return self[readout_key].feature_l1() * self.gamma_readout
-
-    @property
-    def positive(self):
-        if hasattr(self, "_positive"):
-            return self._positive
-        else:
-            return False
-
-    @positive.setter
-    def positive(self, value):
-        self._positive = value
-        for k in self:
-            self[k].positive = value
+                self.add_module(data_key, ClonedReadout(self[original_readout]))
 
+        self.initialize(mean_activity_dict)
 
-class MultiplePointPyramid2d(MultiReadout):
-    _base_readout = PointPyramid2d
+    def prepare_readout_kwargs(self, i, data_key, first_data_key, **kwargs):
+        return kwargs
```

Comment:
This method seems redundant. Given the current functionality, is there a reason it's there? 
On the other hand, I feel like it should do more: get the `kwargs`, compare them to the arguments of the `_base_readout`, and then pass the relevant arguments with a warning, or raise an error.

Reply:
It's just there so that it can be defined in a more meaningful way in `MultiReadoutSharedParametersBase`